internal/controller/compaction/reconciler.go

```diff
@@ -38,6 +38,10 @@ import (
 const (
 	// DefaultETCDQuota is the default etcd quota.
 	DefaultETCDQuota = 8 * 1024 * 1024 * 1024 // 8Gi
+
+	// SafeToEvictKey - annotation that ignores constraints to evict a pod like not being replicated, being on
+	// kube-system namespace or having a local storage if set to "false".
+	SafeToEvictKey = "cluster-autoscaler.kubernetes.io/safe-to-evict"
```

Comment:
Druid should not have any idea about which annotations are being set. Druid should simply never set the Etcd `spec.annotations` on the compaction job pod, because that is something the creator of the Etcd resource has no idea about. I strongly recommend completely removing the logic of adding spec.annotations to the compaction job pod.

Reply:
After an out-of-band discussion with @ishan16696 and @unmarshall , we decided to go ahead with this change, since the current Etcd API does not provide any way for users to specify separate set of annotations for the compaction jobs. We will remove this special handling when working on #580